src/hotspot/cpu/x86/stubGenerator_x86_64_aes.cpp

```diff
@@ -2730,65 +2761,115 @@ void StubGenerator::generateHtbl_48_block_zmm(Register htbl, Register avx512_htb
+  __ evpclmulqdq(ZT1, ZT7, xmm11, 0x10, Assembler::AVX_512bit);
+  __ vpshufd(ZT2, ZT7, 78, Assembler::AVX_512bit);
+  __ evpxorq(ZT1, ZT1, ZT2, Assembler::AVX_512bit);
+  __ evmovdquq(Address(avx512_htbl, 16 * 60), ZT1, Assembler::AVX_512bit);
+  //**ZT1 amd ZT2 to be cleared(hash key)
+
+  //switch to 4x128 - bit computations now
+  __ evshufi64x2(ZT5, ZT5, ZT5, 0x00, Assembler::AVX_512bit); //;; broadcast HashKey ^ 4 across all ZT5
+  __ evmovdquq(ZT8, ZT7, Assembler::AVX_512bit);//; save HashKey ^ 4 to HashKey ^ 1 in ZT8
+  //**ZT8 to be cleared(hash key)
+
+  //calculate HashKey ^ 5 << 1 mod poly, HashKey ^ 6 << 1 mod poly, ... HashKey ^ 8 << 1 mod poly
+  gfmul_avx512(ZT7, ZT5);
+  __ evmovdquq(Address(avx512_htbl, 16 * 24), ZT7, Assembler::AVX_512bit);//; HashKey ^ 8 to HashKey ^ 5 in ZT7 now
 
-  gfmul_avx512(ZT5, HK);
-  __ movdqu(Address(avx512_htbl, 16 * 45), ZT5); // H ^ 2 * 3
-  __ vinserti32x4(ZT7, ZT7, ZT5, 1);
+  //calculate HashKeyX = HashKey x POLY
+  __ evpclmulqdq(ZT1, ZT7, xmm11, 0x10, Assembler::AVX_512bit);
+  __ vpshufd(ZT2, ZT7, 78, Assembler::AVX_512bit);
+  __ evpxorq(ZT1, ZT1, ZT2, Assembler::AVX_512bit);
+  __ evmovdquq(Address(avx512_htbl, 16 * 56), ZT1, Assembler::AVX_512bit);
 
-  gfmul_avx512(ZT5, HK);
-  __ movdqu(Address(avx512_htbl, 16 * 44), ZT5); // H ^ 2 * 4
-  __ vinserti32x4(ZT7, ZT7, ZT5, 0);
+  __ evshufi64x2(ZT5, ZT7, ZT7, 0x00, Assembler::AVX_512bit);//;; broadcast HashKey ^ 8 across all ZT5
 
-  __ evshufi64x2(ZT5, ZT5, ZT5, 0x00, Assembler::AVX_512bit);
-  __ evmovdquq(ZT8, ZT7, Assembler::AVX_512bit);
-  gfmul_avx512(ZT7, ZT5);
-  __ evmovdquq(Address(avx512_htbl, 16 * 40), ZT7, Assembler::AVX_512bit);
-  __ evshufi64x2(ZT5, ZT7, ZT7, 0x00, Assembler::AVX_512bit);
-  gfmul_avx512(ZT8, ZT5);
-  __ evmovdquq(Address(avx512_htbl, 16 * 36), ZT8, Assembler::AVX_512bit);
-  gfmul_avx512(ZT7, ZT5);
-  __ evmovdquq(Address(avx512_htbl, 16 * 32), ZT7, Assembler::AVX_512bit);
-  gfmul_avx512(ZT8, ZT5);
-  __ evmovdquq(Address(avx512_htbl, 16 * 28), ZT8, Assembler::AVX_512bit);
-  gfmul_avx512(ZT7, ZT5);
-  __ evmovdquq(Address(avx512_htbl, 16 * 24), ZT7, Assembler::AVX_512bit);
+  //calculate HashKey ^ 9 << 1 mod poly, HashKey ^ 10 << 1 mod poly, ... HashKey ^ 32 << 1 mod poly
+  //use HashKey ^ 8 as multiplier against ZT8 and ZT7 - this allows deeper ooo execution
+  //compute HashKey ^ (4 + n), HashKey ^ (3 + n), ... HashKey ^ (1 + n)
   gfmul_avx512(ZT8, ZT5);
   __ evmovdquq(Address(avx512_htbl, 16 * 20), ZT8, Assembler::AVX_512bit);
+
+  //calculate HashKeyK = HashKey x POLY
+  __ evpclmulqdq(ZT1, ZT8, xmm11, 0x10, Assembler::AVX_512bit);
+  __ vpshufd(ZT2, ZT8, 78, Assembler::AVX_512bit);
+  __ evpxorq(ZT1, ZT1, ZT2, Assembler::AVX_512bit);
+  __ evmovdquq(Address(avx512_htbl, 16 * 52), ZT1, Assembler::AVX_512bit);
+
+  //compute HashKey ^ (8 + n), HashKey ^ (7 + n), ... HashKey ^ (5 + n)
   gfmul_avx512(ZT7, ZT5);
   __ evmovdquq(Address(avx512_htbl, 16 * 16), ZT7, Assembler::AVX_512bit);
```

Comment:
You can define a macro for this repetitive sequence and pass varying argument to it for brevity.